src/services/findAllReferences.ts

```diff
@@ -2014,16 +2030,41 @@ namespace ts.FindAllReferences {
             readonly symbol: Symbol;
             readonly kind: NodeEntryKind | undefined;
         }
+
+        function isStatic(symbol: Symbol): boolean {
+            if (!symbol.valueDeclaration) { return false; }
+            if (symbol.valueDeclaration?.modifierFlagsCache) { return !!((symbol.valueDeclaration.modifierFlagsCache & ModifierFlags.Static)); }
+            return !!forEach(symbol.valueDeclaration.modifiers, modifier => modifier.kind & ModifierFlags.Static);
+        }
+
         function getRelatedSymbol(search: Search, referenceSymbol: Symbol, referenceLocation: Node, state: State): RelatedSymbol | undefined {
             const { checker } = state;
+            const isReferenceSymbolStatic = isStatic(referenceSymbol);
+            function cbSymbol(sym: Symbol, rootSymbol: Symbol, baseSymbol: Symbol | undefined, kind: NodeEntryKind): RelatedSymbol | undefined {
+                // check whether the symbol used to search itself is just the searched one.
+                if (baseSymbol) {
+                    // static method/property and instance method/property might have the same name. Only check static or only check instance.
+                    if ((isReferenceSymbolStatic && !isStatic(baseSymbol)) || (!isReferenceSymbolStatic && isStatic(baseSymbol))) {

```

Comment:
why not just check isReferecedSymbolStatic !== isStatic(baseSymbol)

Reply:
Whooops, I forget to check other codes, sorry for that!